server/internal/thirdparty/polar/client.go

```diff
@@ -69,9 +74,37 @@ func NewClient(polarClient *polargo.Polar, logger *slog.Logger, tracerProvider t
 		customerStateCache: cache.NewTypedObjectCache[PolarCustomerState](logger.With(attr.SlogCacheNamespace("polar-customer-state")), cache.NewRedisCacheAdapter(redisClient), cache.SuffixNone),
 		productCache:       cache.NewTypedObjectCache[Product](logger.With(attr.SlogCacheNamespace("polar-product")), cache.NewRedisCacheAdapter(redisClient), cache.SuffixNone),
 		periodUsageStorage: cache.NewTypedObjectCache[PolarPeriodUsageState](logger.With(attr.SlogCacheNamespace("polar-period-usage-state")), cache.NewRedisCacheAdapter(redisClient), cache.SuffixNone),
+		webhookSecret:      webhookSecret,
 	}
 }
 
+func (p *Client) ValidateAndParseWebhookEvent(ctx context.Context, payload []byte, webhookHeader http.Header) (*billing.PolarWebhookPayload, error) {
+	base64Secret := base64.StdEncoding.EncodeToString([]byte(p.webhookSecret))
+	wh, err := standardwebhooks.NewWebhook(base64Secret)
+	if err != nil {
+		return nil, fmt.Errorf("create webhook verifier: %w", err)
+	}
+
+	if err := wh.Verify(payload, webhookHeader); err != nil {
+		return nil, fmt.Errorf("verify webhook: %w", err)
+	}
+
+	var webhookPayload billing.PolarWebhookPayload
+	if err := json.Unmarshal(payload, &webhookPayload); err != nil {
+		return nil, fmt.Errorf("unmarshal webhook payload: %w", err)
+	}
+
+	return &webhookPayload, nil
+}
+
+func (p *Client) InvalidateCustomerTierCache(ctx context.Context, orgID string) error {
```

Comment:
Do we need to invalidate more than just this? Also the name isn't quite correct I dont think

Reply:
What else do we need to invalidate. PeriodUsage?